File: src/main/java/com/bdev/hogwarts_api/service/student_attendance/StudentAttendanceService.java
package com.bdev.hogwarts_api.service.student_attendance;

import com.bdev.hogwarts_api.data.dto.student.StudentAttendance;

import java.util.List;

public interface StudentAttendanceService {
    boolean exists(long attendanceId);
    List<StudentAttendance> getAllAttendances();
    List<StudentAttendance> getAttendances(long studentId);
    long addAttendance(StudentAttendance attendance);
    void deleteAttendance(long attendanceId);
}

File: src/main/java/com/bdev/hogwarts_api/service/student_attendance/StudentAttendanceService.kt
package com.bdev.hogwarts_api.service.student_attendance

import com.bdev.hogwarts_api.data.dto.student.StudentAttendance
import com.bdev.hogwarts_api.data.dto.student.StudentAttendanceId

interface StudentAttendanceService {
    fun getAllAttendances(): List<StudentAttendance>
    fun getStudentAttendances(studentLogin: String): List<StudentAttendance>
    fun addAttendance(attendance: StudentAttendance)
    fun deleteAttendance(attendanceId: StudentAttendanceId)
}
